src/main/java/com/ita/if103java/ims/dao/impl/DashboardDaoImpl.java

```diff
@@ -102,50 +102,50 @@ public WarehousePremiumStructDto getPreLoadByAccounId(Long id, Long accountId) {
                 warehousePremiumStructRowMapper, id, accountId);
-                for (int i = 0; i < wpld.size(); i++) {
-                    wpld.get(i).setLevel(level + 1);
-
-                    ChargeCapacity cc1 = jdbcTemplate.queryForObject(Queries.SQL_FIND_SUBS_LOAD,
-                        chargeCapacityRowMapper,
-                        wpld.get(i).getId(), accountId, accountId, accountId, accountId);
-
-                    wpld.get(i).setCharge(cc1.getCharge());
-                    wpld.get(i).setCapacity(cc1.getCapacity());
-
-                    wpld.get(i).setChilds(findSub(wpld.get(i).getId(), wpld.get(i).getLevel(), accountId));
-                    if (wpld.get(i).getChilds().size() == 0){
-                        ChargeCapacity cc2 = jdbcTemplate.queryForObject(Queries.SQL_FIND_BOT_LOAD,
-                            chargeCapacityRowMapper, wpld.get(i).getId(), accountId);
-
-                        wpld.get(i).setCharge(cc2.getCharge());
-                        wpld.get(i).setCapacity(cc2.getCapacity());
-                    }
+            for (int i = 0; i < wpld.size(); i++) {
+                wpld.get(i).setLevel(level + 1);
+                wpld.get(i).setChilds(findSub(wpld.get(i).getId(), top_warehouse_id, wpld.get(i).getLevel(), accountId));
+                if (wpld.get(i).getChilds().size()==0){
+                    ChargeCapacity chargeCapacity = jdbcTemplate.queryForObject(Queries.SQL_FIND_BOT_CAPACITY_CHARGE,
+                        chargeCapacityRowMapper, top_warehouse_id, wpld.get(i).getId(), accountId);
+                    wpld.get(i).setCharge(chargeCapacity.getCharge());
+                    wpld.get(i).setCapacity(chargeCapacity.getCapacity());
                 }
-                return wpld;
+            }
+            return wpld;
         } catch (DataAccessException e) {
-            throw crudException(e.toString(),"findSub","*");
+            throw crudException(e,"findSub","*");
+        }
+    }
+
+    private void recFill(WarehousePremiumStructDto wp){
+        for(int i=0; i < wp.getChilds().size(); i++){
+            recFill(wp.getChilds().get(i));
+        }
+        for(int i=0; i<wp.getChilds().size(); i++){
+            wp.setCharge(wp.getCharge() + wp.getChilds().get(i).getCharge());
+            wp.setCapacity(wp.getCapacity() + wp.getChilds().get(i).getCapacity());
         }
     }
 
-    private CRUDException crudException(String message, String operation, String attribute) {
-        CRUDException exception = new CRUDException(message);
-        LOGGER.error("CRUDException exception. Operation:({}) Dashboard ({}) exception. Message: {}", operation, attribute, message);
+    private CRUDException crudException(Exception e, String operation, String attribute) {
+        CRUDException exception = new CRUDException(e);
+        LOGGER.error("CRUDException exception. Operation:({}) Dashboard ({}) exception.", operation, attribute, e);
```

Comment:
Replace this:
```
CRUDException exception = new CRUDException(e);
LOGGER.error("CRUDException exception. Operation:({}) Dashboard ({}) exception.", operation, attribute, e);
```
with this:
```
CRUDException exception = new CRUDException("CRUDException exception. Operation:(" + operation + ") Attribute (" + attribute + ").", e);
```
No need to log here as it will be logged in the GlobalExceptionHandler